internal/chain/engine.go

```diff
@@ -725,7 +725,8 @@ func readShardState(chain engine.ChainReader, epoch *big.Int, targetShardID uint
 		return shardState, nil
 
 	} else {
-		shardState, err := chain.ReadShardState(epoch)
+		//shardState, err := chain.ReadShardState(epoch)
+		shardState, err := committee.WithStakingEnabled.ReadFromDB(epoch, chain)
```

Comment:
why doing this change?

Reply:
The two are equivalent, but since`committee.WithStakingEnabled.Compute` is used in the IF-TRUE branch, I think we should use `committee.WithStakingEnabled.ReadFromDB` in the else branch.
Of course, we can also keep it unchanged here.